File: src/main/java/io/zeitwert/fm/building/service/api/BuildingService.java
package io.zeitwert.fm.building.service.api;

import com.google.maps.ImageResult;
import com.google.maps.model.GeocodingResult;
import com.google.maps.model.Size;

public interface BuildingService {

	String getCoordinates(String address);

	GeocodingResult getGeocoding(String address);

	ImageResult getMap(String name, String address, Size size, int zoom);

}

File: fm-domain/src/main/java/io/zeitwert/fm/building/service/api/BuildingService.kt
package io.zeitwert.fm.building.service.api

import com.google.maps.ImageResult
import com.google.maps.model.GeocodingResult
import com.google.maps.model.Size

interface BuildingService {

	fun getCoordinates(address: String): String?

	fun getGeocoding(address: String): GeocodingResult?

	fun getMap(address: String, size: Size, zoom: Int): ImageResult?

}
